plugins/field-angle/src/field_angle.ts

```diff
@@ -0,0 +1,553 @@
+   */
+  private round = FieldAngle.ROUND;
+
+  /**
+   * Array holding info needed to unbind events.
+   * Used for disposing.
+   * Ex: [[node, name, func], [node, name, func]].
+   */
+  private boundEvents: Blockly.browserEvents.Data[] = [];
+
+  /** Dynamic red line pointing at the value's angle. */
+  private line: SVGLineElement|null = null;
+
+  /** Dynamic pink area extending from 0 to the value's angle. */
+  private gauge: SVGPathElement|null = null;
+
+  /** The degree symbol for this field. */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  protected symbol_: SVGTSpanElement|null = null;
+
+  /**
+   * @param value The initial value of the field.  Should cast to a number.
+   *     Defaults to 0.  Also accepts Field.SKIP_SETUP if you wish to skip setup
+   *     (only used by subclasses that want to handle configuration and setting
+   *     the field value after their own constructors have run).
+   * @param validator A function that is called to validate changes to the
+   *     field's value.  Takes in a number & returns a validated number, or null
+   *     to abort the change.
+   * @param config A map of options used to configure the field.
+   *     See the [field creation documentation]{@link
+   * https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/angle#creation}
+   * for a list of properties this parameter supports.
+   */
+  constructor(value?: string|number|typeof Blockly.Field.SKIP_SETUP,
+      validator?: FieldAngleValidator, config?: FieldAngleConfig) {
+    super(Blockly.Field.SKIP_SETUP);
+
+    if (value === Blockly.Field.SKIP_SETUP) return;
+    if (config) {
+      this.configure_(config);
+    }
+    this.setValue(value);
+    if (validator) {
+      this.setValidator(validator);
+    }
+  }
+
+  /**
+   * Configure the field based on the given map of options.
+   *
+   * @param config A map of options to configure the field based on.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  protected override configure_(config: FieldAngleConfig) {
+    super.configure_(config);
+    switch (config.mode) {
+      case Mode.COMPASS:
+        this.clockwise = true;
+        this.offset = 90;
+        break;
+      case Mode.PROTRACTOR:
+        // This is the default mode, so we could do nothing.  But just to
+        // future-proof, we'll set it anyway.
+        this.clockwise = false;
+        this.offset = 0;
+        break;
+    }
+
+    // Allow individual settings to override the mode setting.
+    if (config.clockwise) this.clockwise = config.clockwise;
+    if (config.offset) this.offset = config.offset;
+    if (config.wrap) this.wrap = config.wrap;
+    if (config.round) this.round = config.round;
+  }
+
+  /**
+   * Create the block UI for this field.
+   *
+   * @internal
+   */
+  override initView() {
+    super.initView();
+    // Add the degree symbol to the left of the number,
+    // even in RTL (issue #2380).
```

Comment:
Change to a full link since it's going into a new repository: https://github.com/google/blockly/issues/2380